src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -496,13 +496,24 @@ ParsedQuery Visitor::visit(Parser::ModifyContext* ctx) {
       return true;
     }
   };
+  auto isVisibleIfVariableGraph =
+      [this](const SparqlTripleSimpleWithGraph::Graph& graph) {
+        if (std::holds_alternative<Variable>(graph)) {
+          return std::ranges::find(parsedQuery_.getVisibleVariables(),
+                                   std::get<Variable>(graph)) !=
+                 parsedQuery_.getVisibleVariables().end();
+        } else {
+          return true;
+        }
+      };
   auto checkTriples =
-      [&isVisibleIfVariable,
-       &ctx](const std::vector<SparqlTripleSimpleWithGraph>& triples) {
+      [&isVisibleIfVariable, &ctx, &isVisibleIfVariableGraph](
+          const std::vector<SparqlTripleSimpleWithGraph>& triples) {
         for (auto& triple : triples) {
           if (!(isVisibleIfVariable(triple.s_) &&
                 isVisibleIfVariable(triple.p_) &&
-                isVisibleIfVariable(triple.o_))) {
+                isVisibleIfVariable(triple.o_) &&
```

Comment:
This (and the corresponding tests) looks completely unrelated and should be in a separate PR.